### PR TITLE
New/settings page

### PR DESCRIPTION
**Pull request by vb-oiko**

### CODE REVIEW

File: client/src/scenes/SettingsProfile/index.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  imgEditBtn = (
+    <Button secondary compact size="tiny">
+        <Icon name="edit" />
+      Edit
+    </Button>
+  );
+
+  render() {
+      return (
+          <div>
```
Comment (TheSubliminal) [inline client/src/scenes/SettingsProfile/index.jsx]: Если нету никаких стилей связанных с этим div-ом, то можно заменить
```suggestion
          <>
```

File: client/src/scenes/SettingsProfile/index.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                  <Grid.Column computer={12} tablet={16} mobile={16}>
+                      <Header as="h2">Public profile</Header>
+                      <Divider />
+                      <Grid>
+                          <Grid.Column computer={11} tablet={16} mobile={16}>
+                              <Formik initialValues={this.props.currentUser} onSubmit={this.onSubmit}>
+                                  {({ isSubmitting }) => (
+                                      <Form className="ui form">
+                                          <Header as="h4">Name</Header>
+                                          <Field type="text" name="name" render={this.renderField} />
```
Comment (alexandrtovmach) [inline client/src/scenes/SettingsProfile/index.jsx]: What about validation and errors?
https://jaredpalmer.com/formik/docs/api/errormessage

File: client/src/scenes/SettingsProfile/index.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                          <Menu.Item header>Personal settings</Menu.Item>
+                          <Menu.Item>Profile</Menu.Item>
+                      </Menu>
+                  </Grid.Column>
+                  <Grid.Column computer={12} tablet={16} mobile={16}>
+                      <Header as="h2">Public profile</Header>
+                      <Divider />
+                      <Grid>
+                          <Grid.Column computer={11} tablet={16} mobile={16}>
+                              <Formik initialValues={this.props.currentUser} onSubmit={this.onSubmit}>
```
Comment (alexandrtovmach) [inline client/src/scenes/SettingsProfile/index.jsx]: Use destructurization for variables like this: `this.props.currentUser`

File: client/src/scenes/SettingsProfile/index.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  }
+}
+
+SettingsProfile.propTypes = {
+    currentUser: PropTypes.object,
+    updateUserSettings: PropTypes.func
+};
+
+const mapStateToProps = state => {
+    console.warn(state.profile);
```
Comment (alexandrtovmach) [inline client/src/scenes/SettingsProfile/index.jsx]: Remove unusable logs

File: client/src/scenes/SettingsProfile/notes.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Comment (alexandrtovmach) [inline client/src/scenes/SettingsProfile/notes.js]: Not sure about reasonability of this file. This content can be easily placed in https://github.com/BinaryStudioAcademy/bsa-2019-depot/pull/51/files#diff-da9d18a2494157ea57cc7c8437adc8dd

File: client/src/scenes/SettingsProfile/index.jsx (hunk 2 above)
Comment (vb-oiko) [reply client/src/scenes/SettingsProfile/index.jsx]: It's about first, last etc. name
Not about username
No validation is needed here

File: client/src/scenes/SettingsProfile/index.jsx (hunk 3 above)
Comment (vb-oiko) [reply client/src/scenes/SettingsProfile/index.jsx]: нsorry, can't understand what should I do

File: client/src/scenes/SettingsProfile/index.jsx (hunk 4 above)
Comment (vb-oiko) [reply client/src/scenes/SettingsProfile/index.jsx]: done

File: client/src/scenes/SettingsProfile/notes.js (hunk 5 above)
Comment (vb-oiko) [reply client/src/scenes/SettingsProfile/notes.js]: done

File: client/src/scenes/SettingsProfile/index.jsx (hunk 3 above)
Comment (alexandrtovmach) [reply client/src/scenes/SettingsProfile/index.jsx]:
```
const { currentUser: { loading } } = this.props;
```

Comment (vb-oiko) [reply client/src/scenes/SettingsProfile/index.jsx]: done

**Comment by TheSubliminal** [comment]
Looks good, but conflicts need to be resolved

**Pull request merged**